lib/private/Collaboration/Collaborators/Search.php

```diff
@@ -75,7 +75,7 @@ public function search($search, array $shareTypes, $lookup, $limit, $offset) {
 		// Get from lookup server, not a separate share type
 		if ($lookup) {
 			$searchPlugin = $this->c->resolve(LookupPlugin::class);
-			$hasMoreResults |= $searchPlugin->search($search, $limit, $offset, $searchResult);
+			$hasMoreResults = $searchPlugin->search($search, $limit, $offset, $searchResult);
```

Comment:
this discards the result of possible previous search operations

Reply:
check the code. `$hasMoreResults` is initialized with `false` and `|` is a numeric operator. This was a bug.

Reply:
contra.

it is a bitwise or, yes, and it returns an int, yes, and yet it works with bool still. The correct approach would then be

```suggestion
			$hasMoreResults = $hasMoreResults || $searchPlugin->search($search, $limit, $offset, $searchResult);
```

instead of introducing a bug :)

Reply:
Oh I found where my train of though derailed: I didn't see we're in a loop there. Makes sense. Let me change to this.